http/src/main/java/io/micronaut/http/cookie/DefaultServerCookieDecoder.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.cookie;
+
+import io.micronaut.core.annotation.NonNull;
+
+import java.net.HttpCookie;
+import java.util.List;
+
+/**
+ * {@link ServerCookieDecoder} implementation backed by {@link HttpCookie#parse(String)}.
+ *
+ * @author Sergio del Amo
+ * @since 4.3.0
+ */
+public class DefaultServerCookieDecoder implements ServerCookieDecoder {
```

Comment:
Can this be `final` and `@Internal`? Does it need to be public?

Reply:
> can this be `final`? 

https://github.com/micronaut-projects/micronaut-core/pull/10402/commits/a8d3204d9c512fcc50950358fd14bf25905b6bbe

> can this be `@Internal`? 
https://github.com/micronaut-projects/micronaut-core/pull/10402/commits/ce14d1e942177abd7c82c437888b4d283759009a

> Does it need to be public?

Yes, it needs to be public.